Community/Tdarr_Plugin_goof1_URL_Plex_Refresh.js

```diff
@@ -0,0 +1,221 @@
+      type: 'string',
+      defaultValue: '',
+      inputUI: {
+        type: 'text',
+      },
+      tooltip: `
+              If the Plex path is not the same as the local path you may need to sub parts of the path. 
+              Here is where you would enter the path that Plex uses to find the file. You would only enter the part of the path that is different.
+
+              If your TDarr path is:
+              /media/local/tv/The Best Show Evaaaarr/Season 2/The Best Show Evaaaarr - S02E31 - Heck Yea HDTV-720p.mp4
+              
+              And the Plex path to the file is:
+              /data/tv/The Best Show Evaaaarr/Season 2/The Best Show Evaaaarr - S02E31 - Heck Yea HDTV-720p.mp4
+
+              then part you would enter here is:
+
+               \\nExample:\\n
+               /data/`,
+    },
+    {
+      name: 'TDarr_Path',
+      type: 'string',
+      defaultValue: '',
+      inputUI: {
+        type: 'text',
+      },
+      tooltip: `
+              If the Plex path is not the same as the local path you may need to sub parts of the path. 
+              Here is where you would enter the path that Plex uses to find the file. You would only enter the part of the path that is different.
+
+              If your TDarr path is:
+              /media/local/tv/The Best Show Evaaaarr/Season 2/The Best Show Evaaaarr - S02E31 - Heck Yea HDTV-720p.mp4
+              
+              And the Plex path to the file is:
+              /data/tv/The Best Show Evaaaarr/Season 2/The Best Show Evaaaarr - S02E31 - Heck Yea HDTV-720p.mp4
+
+              then part you would enter here is:
+
+               \\nExample:\\n
+               /media/local/`,
+    },
+  ],
+});
+
+const plugin = (file, librarySettings, inputs, otherArguments) => {
+  inputs = loadDefaultValues(inputs, details);
+
+  let response = {
+    file,
+    removeFromDB: false,
+    updateDB: false,
+    infoLog: '',
+  };
+
+  const http = require('http')
+  const https = require('https')
+
+  const type = inputs.Url_Protocol
+  const url = inputs.Plex_Url
+  const port = inputs.Plex_Port
+  const token = inputs.Plex_Token
+  const key = inputs.Library_Key
+  const plexPath = inputs.Plex_Path
+  const tdarrPath = inputs.TDarr_Path
+
+  if (!type || !url || !token || !key) {
+    response.infoLog = `Url_Protocol, Plex_Url, Plex_Token, and Library_Key are all required`
+    response.processFile = false
+    return response
+  }
+
+
+  let filePath = file.file
+  filePath = filePath.substring(0, filePath.lastIndexOf("/"));
+
+  if (tdarrPath || plexPath) {
+    filePath = filePath.replace(tdarrPath, plexPath)
+  }
+
+  response.infoLog += `Attempting to update Plex path ${filePath} in library ${key}\n`
+
+  const fullUrl = `${type}://${url}${port ? `:${port}` : ''}/library/sections/${key}/refresh?path=${filePath}&X-Plex-Token=${token}`
+  const redactedUrl = `${type}://${url}${port ? `:${port}` : ''}/library/sections/${key}/refresh?path=${filePath}&X-Plex-Token=[redacted]`
```

Comment:
You could build this as a sort of `tempUrl` variable and only append the token at the request. Now you're *basically* doing the same thing twice, just with the token or without.

Move the ternary out of the string itself to a separate variable. Increases readability.